tools/doctor

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# Peforms basic checks for tools required by Magic Modules.
+
+set -e
+
+if [ -t 1 ]; then
+    red=$'\e[1;31m'
+    grn=$'\e[1;32m'
+    yel=$'\e[1;33m'
+    blu=$'\e[1;34m'
+    mag=$'\e[1;35m'
+    cyn=$'\e[1;36m'
+    end=$'\e[0m'
+else
+    red=''
+    grn=''
+    yel=''
+    blu=''
+    mag=''
+    cyn=''
+    end=''
+fi
+
+warn() {
+    printf "%s\n" "${!1}${2}${end}"
+}
+
+found=0
+count_in_path() {
+    echo "Check for ${1} in path..."
+    found=$(which -a "${1}" | uniq | wc -l)
+}
+
+count_in_path "rbenv"
+if [ $found -eq 0 ]; then
+    warn "yel" "   rbenv not found. See https://github.com/rbenv/rbenv for information on installing it."
```

Comment:
Looks like every call to `warn` takes in a string with three empty spaces in the front. Could we move those spaces in to the `warn` function?

Reply:
Done.